lightning/src/routing/network_graph.rs

```diff
@@ -127,29 +129,33 @@ impl<C: Deref + Sync + Send, L: Deref + Sync + Send> RoutingMessageHandler for N
 			return Err(LightningError{err: "Channel announcement node had a channel with itself".to_owned(), action: ErrorAction::IgnoreError});
 		}
 
-		let utxo_value = match self.chain_monitor.get_chain_utxo(msg.contents.chain_hash, msg.contents.short_channel_id) {
-			Ok((script_pubkey, value)) => {
-				let expected_script = Builder::new().push_opcode(opcodes::all::OP_PUSHNUM_2)
-				                                    .push_slice(&msg.contents.bitcoin_key_1.serialize())
-				                                    .push_slice(&msg.contents.bitcoin_key_2.serialize())
-				                                    .push_opcode(opcodes::all::OP_PUSHNUM_2)
-				                                    .push_opcode(opcodes::all::OP_CHECKMULTISIG).into_script().to_v0_p2wsh();
-				if script_pubkey != expected_script {
-					return Err(LightningError{err: format!("Channel announcement key ({}) didn't match on-chain script ({})", script_pubkey.to_hex(), expected_script.to_hex()), action: ErrorAction::IgnoreError});
-				}
-				//TODO: Check if value is worth storing, use it to inform routing, and compare it
-				//to the new HTLC max field in channel_update
-				Some(value)
-			},
-			Err(ChainError::NotSupported) => {
+		let utxo_value = match &self.chain_access {
+			&None => {
 				// Tentatively accept, potentially exposing us to DoS attacks
 				None
 			},
-			Err(ChainError::NotWatched) => {
-				return Err(LightningError{err: format!("Channel announced on an unknown chain ({})", msg.contents.chain_hash.encode().to_hex()), action: ErrorAction::IgnoreError});
-			},
-			Err(ChainError::UnknownTx) => {
-				return Err(LightningError{err: "Channel announced without corresponding UTXO entry".to_owned(), action: ErrorAction::IgnoreError});
+			&Some(ref chain_access) => {
+				match chain_access.get_utxo(&msg.contents.chain_hash, msg.contents.short_channel_id) {
+					Ok(TxOut { value, script_pubkey }) => {
+						let expected_script = Builder::new().push_opcode(opcodes::all::OP_PUSHNUM_2)
```

Comment:
An easy follow-up `get_funding_transaction_redeemscript()` to improve readability.